### PR TITLE
List, List item and List item text initial code 

### PR DESCRIPTION
**Pull request by RichardGarcia**

### CODE REVIEW

**Comment by agentlewis** [comment]
I think we should make list item and list item text the same atom? to me they are the same atom much like a button text and area are the same atom.

File: ui/components/atoms/A012-ListItem/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import ListItem from '@material-ui/core/ListItem'
+
+import schema from './schema'
+
+const A012 = props => {
+  const {button, children} = props
+
+  return (
+    <ListItem button={button}>
```
Comment (agentlewis) [inline ui/components/atoms/A012-ListItem/index.js]: for the purposes of the mobile navigation list we want to make I think we should hardcode this button as true ie not assign a value to it.

**Review by danalexilewis — CHANGES_REQUESTED** [review]
Lets collapse list item and list item text into one atom and make it always be a button

File: ui/components/atoms/A012-ListItem/index.js (hunk 1 above)
Comment (RichardGarcia) [reply ui/components/atoms/A012-ListItem/index.js]: updated ListItem

**Comment by RichardGarcia** [comment]
> I think we should make list item and list item text the same atom? to me they are the same atom much like a button text and area are the same atom.

Dan, I don't think its possible List Item and List Item Text in the same icon. List Item is like a container for List Item Text and other element / Atom like Icons, Avatar

**Comment by agentlewis** [comment]
I’m pretty sure we could do something like this:

```
const {iconname, text} = props
<Listbutton>
  <ListItem>
    {iconname&&<Icon iconname={iconname}/>}
    <ListItemText primary={text}/>
  </ListItem>
</ListButton>
```

You may have to check that Conditional render with a lodash exists function - ie it shouldn’t be shown if it’s iconname is null, undefined or an empty string.

Also conditional rendering can be [read about here](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)

How do we position the icon to the right?

**Comment by agentlewis** [comment]
@RichardGarcia hey I just realised what you meant by
> List Item is like a container for List Item Text and other element / Atom like Icons, Avatar

Yes i get that - what I want to do is make a list item per type of atom we need. ie I don't want to just replicate the component as you have currently done. I want to do something more like what we did with textfield - making multiple atoms using the same material ui component. In this case we want an ListNavItem - ie a list item that looks like a nav arrow to the right that if you click will open a new list.

(the action/nav function will be passed down as a handleOnClick function)

**Comment by agentlewis** [comment]
@RichardGarcia I finished this off along the lines I wanted. ie more then one list item type. Can you think of anymore we may want?

**Pull request merged**